test/e2e/storage/testsuites/topology.go

```diff
@@ -261,29 +278,39 @@ func topologyEqual(t1, t2 topology) bool {
 	return true
 }
 
-// TODO: how to support drivers with multi-value topologies?
-func (t *topologyTestSuite) setAllowedTopologies(topo topology, sc *storagev1.StorageClass) {
-
-	exprs := []v1.TopologySelectorLabelRequirement{}
-	for k, v := range topo {
-		exprs = append(exprs, v1.TopologySelectorLabelRequirement{
-			Key:    k,
-			Values: []string{v},
-		})
-	}
-
-	sc.AllowedTopologies = []v1.TopologySelectorTerm{
-		{
-			MatchLabelExpressions: exprs,
-		},
+// Set StorageClass.Allowed topologies from topos while excluding the topology at excludedIndex.
+// excludedIndex can be -1 to specify nothing should be excluded.
+// Return the list of allowed topologies generated.
+func (t *topologyTestSuite) setAllowedTopologies(sc *storagev1.StorageClass, topos []topology, excludedIndex int) []topology {
+	allowedTopologies := []topology{}
+	sc.AllowedTopologies = []v1.TopologySelectorTerm{}
+
+	for i := 0; i < len(topos); i++ {
+		if i != excludedIndex {
+			exprs := []v1.TopologySelectorLabelRequirement{}
+			for k, v := range topos[i] {
+				exprs = append(exprs, v1.TopologySelectorLabelRequirement{
+					Key:    k,
+					Values: []string{v},
```

Comment:
We never test the case of multiple values then. Just want to make sure this was intentional. IIRC unit tests cover that case when I last worked on it so it should be OK.

Reply:
Correct, however, you can still test things like RePD by ORing the top level TopologySelectorTerm, which I have done here (and tested).